src/redux/handlers/onMove.js

```diff
@@ -8,24 +9,25 @@ import {
 import {
   switchFirstCardFromBuffer,
   switchCardToBuffer,
-  isPlayerHasCard,
+  doesPlayerHaveCard,
 } from '../../models/player';
 
 import cards from '../../constants/cards';
 import { movePos, isPositionOutOfRange } from '../../models/position';
 import { createMine } from '../../models/mine';
 
 export const onMove = (state, action) => {
-  const card = action.payload.card;
+  const { card } = action.payload;
   let { entities, roundCounter, roundMoves, turn, isEnd, winner } = state;
   let player = entities[turn];
   let players = [];
 
-  if (!isPlayerHasCard(player, card)) {
+  if (!doesPlayerHaveCard(player, card)) {
     throw 'Invalid move';
   }
 
-  player = entities[player.id] = switchCardToBuffer(player, card);
+  Immutable.setIn(state, ['entities', player.id], switchCardToBuffer(player, card));
```

Comment:
ta funkcja nie zwraca nowego stanu przypadkiem? (imm.setIn). w linii 45 przekazujesz entities z początkowego stanu a nie zmienionego

Reply:
Zwraca. Zostawiłem sobie to celowo, razem z // TODO w tej funkcji żebym wiedział na co zwrócić uwagę jak wrócę do tego - dlatego napisałem Ci, że to draft.